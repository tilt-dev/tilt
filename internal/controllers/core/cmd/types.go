package cmd

import "github.com/tilt-dev/tilt/pkg/apis/core/v1alpha1"

type Cmd = v1alpha1.Cmd
type CmdSpec = v1alpha1.CmdSpec
type CmdStatus = v1alpha1.CmdStatus
type CmdStateRunning = v1alpha1.CmdStateRunning
type CmdStateTerminated = v1alpha1.CmdStateTerminated

const LabelManifest = v1alpha1.LabelManifest
