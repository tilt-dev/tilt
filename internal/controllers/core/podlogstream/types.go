package podlogstream

import (
	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"

	"github.com/tilt-dev/tilt/pkg/apis/core/v1alpha1"
)

type PodLogStream = v1alpha1.PodLogStream
type PodLogStreamSpec = v1alpha1.PodLogStreamSpec
type PodLogStreamStatus = v1alpha1.PodLogStreamStatus
type ContainerLogStreamStatus = v1alpha1.ContainerLogStreamStatus
type ObjectMeta = metav1.ObjectMeta
