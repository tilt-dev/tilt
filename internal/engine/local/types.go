package local

import (
	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"

	"github.com/tilt-dev/tilt/pkg/apis/core/v1alpha1"
)

type Cmd = v1alpha1.Cmd
type CmdList = v1alpha1.CmdList
type CmdStatus = v1alpha1.CmdStatus
type CmdSpec = v1alpha1.CmdSpec
type CmdStateWaiting = v1alpha1.CmdStateWaiting
type CmdStateTerminated = v1alpha1.CmdStateTerminated
type CmdStateRunning = v1alpha1.CmdStateRunning
type ObjectMeta = metav1.ObjectMeta
