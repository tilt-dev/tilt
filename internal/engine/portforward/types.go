package portforward

import (
	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"

	"github.com/tilt-dev/tilt/pkg/apis/core/v1alpha1"
)

type PortForward = v1alpha1.PortForward
type PortForwardSpec = v1alpha1.PortForwardSpec
type PortForwardStatus = v1alpha1.PortForwardStatus
type ObjectMeta = metav1.ObjectMeta
type Forward = v1alpha1.Forward
