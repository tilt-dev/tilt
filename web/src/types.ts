import { Alert } from "./alerts"
import { Facet } from "./facets"

export enum SocketState {
  Loading,
  Reconnecting,
  Closed,
  Active,
}

export enum ResourceView {
  Log,
  Alerts,
  Facets = 2,
}

export enum TriggerMode {
  TriggerModeAuto,
  TriggerModeManualAfterInitial,
  TriggerModeManualIncludingInitial,
}

// what is the status of the resource in the cluster
export enum RuntimeStatus {
  Ok = "ok",
  Pending = "pending",
  Error = "error",
  NotApplicable = "not_applicable",
}

// What is the status of the resource with respect to Tilt
export enum ResourceStatus {
  Building, // Tilt is actively doing something (e.g., docker build or kubectl apply)
  Pending, // not building, healthy, or unhealthy, but presumably on its way to one of those (e.g., queued to build, or ContainerCreating)
  Healthy, // e.g., build succeeded and pod is running and healthy
  Unhealthy, // e.g., last build failed, or CrashLoopBackOff
  Warning, // e.g., an undismissed restart
  None, // e.g., a manual build that has never executed
}

export type SnapshotHighlight = {
  beginningLogID: string
  endingLogID: string
  text: string
}

export enum ShowFatalErrorModal {
  Default,
  Show,
  Hide,
}

export type Snapshot = {
  view: Proto.webviewView
  isSidebarClosed: boolean
  path?: string
  snapshotHighlight?: SnapshotHighlight | null
}

// A plaintext representation of a line of the log,
// with metadata to render it in isolation.
//
// The metadata should be stored as primitive fields
// so that React's default caching behavior will kick in.
export type LogLine = {
  // We assume that 'text' does not contain a newline
  text: string
  manifestName: string
  level: string
}
