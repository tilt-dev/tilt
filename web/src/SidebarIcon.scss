@import "constants";

.SidebarIcon {
  display: flex;
  align-items: center;
  justify-content: center;
  width: $sidebar-collapsed-width;
}
