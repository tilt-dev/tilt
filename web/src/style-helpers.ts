export enum Color {
  // Brand Colors
  green = "#20ba31",
  greenLight = "#70d37b",
  blue = "#03c7d3",
  blueLight = "#5edbe3",
  blueDark = "#007d82",
  red = "#f6685c",
  redLight = "#f7aaa4",
  yellow = "#fcb41e",
  yellowLight = "#fdcf6f",
  white = "#ffffff",

  offWhite = "#eef1f1",
  grayLightest = "#93a1a1", // Solarized base1 (darkest content tone)
  grayLight = "#586e75", // Solarized base01
  gray = "#073642", // Solarized base02
  grayDark = "#002b36", // Solarized base03 (darkest bg tone)
  grayDarker = "#00242d",
  grayDarkest = "#001b20", // Brand
  black = "#000000",

  text = "#073642",
}

export enum ColorAlpha {
  almostTransparent = 0.1,
  translucent = 0.3,
  almostOpaque = 0.7,
}

export function ColorRGBA(hex: string, alpha: number) {
  let r = parseInt(hex.slice(1, 3), 16),
    g = parseInt(hex.slice(3, 5), 16),
    b = parseInt(hex.slice(5, 7), 16)

  return `rgba(${r}, ${g}, ${b}, ${alpha})`
}

export enum Font {
  sansSerif = '"Montserrat", "Open Sans", "Helvetica", "Arial", sans-serif',
  monospace = '"Inconsolata", "Monaco", "Courier New", "Courier", monospace',
}

export enum FontSize {
  largest = "40px",
  large = "26px",
  default = "20px",
  small = "16px",
  smallest = "13px",
}

let unit = 32

export function SizeUnit(multiplier: number) {
  return `${unit * multiplier}px`
}

// Set sizes expressed in pixels:
export enum Height {
  unit = unit,
  statusHeader = unit * 1.8, // The bar at the top with Pod ID and status
  secondaryNav = unit * 1.2,
  secondaryNavLower = unit * 0.8,
  secondaryNavOverlap = unit * -0.2,
  secondaryNavTwoLevel = unit * 1.8,
  sidebarItem = unit * 1.4, // sync with constants.scss > $sidebar-item
  resourceBar = unit * 1.5,
  statusbar = unit * 1.5,
}
export enum Width {
  sidebar = unit * 10.5, // Sync with constants.scss > $sidebar-width
  secondaryNavItem = unit * 5,
  sidebarTriggerButton = unit * 1.4,
  sidebarCollapsed = unit, // sync with constants.scss > $sidebar-collapsed-width
  badge = unit * 0.6,
  smallScreen = 1500,
}

export const mixinHideOnSmallScreen = `
@media screen and (max-width: ${Width.smallScreen}px) {
  display: none;
}`

export enum ZIndex {
  HUDHeader = 500,
}

export enum AnimDuration {
  default = "0.3s",
}
