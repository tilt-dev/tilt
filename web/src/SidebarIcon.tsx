import React, { PureComponent } from "react"
import { ResourceStatus } from "./types"
import {
  AnimDuration,
  Color,
  ColorAlpha,
  ColorRGBA,
  Font,
  FontSize,
  SizeUnit,
} from "./style-helpers"
import styled, { keyframes } from "styled-components"
import { Width } from "./style-helpers"
import { Link } from "react-router-dom"

type SidebarIconProps = {
  status: ResourceStatus
  alertCount: number
}

// For testing
export enum IconType {
  StatusDefault = "default",
  StatusPending = "pending",
  StatusBuilding = "building",
}

let glowWhite = keyframes`
  0% {
    background-color: ${ColorRGBA(Color.white, ColorAlpha.translucent)};
  }
  50% {
    background-color: ${ColorRGBA(Color.white, ColorAlpha.almostTransparent)};
  }
`

let glowDark = keyframes`
  0% {
    background-color: ${ColorRGBA(Color.gray, ColorAlpha.translucent)};
  }
  50% {
    background-color: ${ColorRGBA(Color.gray, ColorAlpha.almostTransparent)};
  }
`

let SidebarIconRoot = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  width: ${Width.sidebarCollapsed * 0.7}px;
  margin-right: ${Width.sidebarCollapsed * 0.3}px;
  transition: background-color ${AnimDuration.default} linear,
    opacity ${AnimDuration.default} linear;

  &.isWarning {
    background-color: ${Color.yellow};
  }
  &.isHealthy {
    background-color: ${Color.green};
  }
  &.isUnhealthy {
    background-color: ${Color.red};
  }
  &.isBuilding {
    background-color: ${ColorRGBA(Color.white, ColorAlpha.translucent)};
  }
  .isSelected &.isBuilding {
    background-color: ${ColorRGBA(Color.gray, ColorAlpha.translucent)};
  }
  &.isPending {
    background-color: ${ColorRGBA(Color.white, ColorAlpha.translucent)};
    animation: ${glowWhite} 2s linear infinite;
  }
  .isSelected &.isPending {
    background-color: ${ColorRGBA(Color.gray, ColorAlpha.translucent)};
    animation: ${glowDark} 2s linear infinite;
  }
  &.isNone {
    background-color: ${ColorRGBA(Color.white, ColorAlpha.translucent)};
  }
  .isSelected &.isNone {
    background-color: ${ColorRGBA(Color.gray, ColorAlpha.translucent)};
  }
`

let AlertCount = styled.span`
  font-family: ${Font.sansSerif};
  font-size: ${FontSize.smallest};
  color: ${Color.black};
`

export default class SidebarIcon extends PureComponent<SidebarIconProps> {
  render() {
    return (
      <SidebarIconRoot className={`${this.status()}`}>
        {this.props.alertCount > 0 ? (
          <AlertCount>{this.props.alertCount}</AlertCount>
        ) : (
          <span>&nbsp;</span>
        )}
      </SidebarIconRoot>
    )
  }

  status() {
    switch (this.props.status) {
      case ResourceStatus.Building:
        return "isBuilding"
      case ResourceStatus.Pending:
        return "isPending"
      case ResourceStatus.Warning:
        return "isWarning"
      case ResourceStatus.Healthy:
        return "isHealthy"
      case ResourceStatus.Unhealthy:
        return "isUnhealthy"
      case ResourceStatus.None:
        return "isNone"
    }
  }
}
