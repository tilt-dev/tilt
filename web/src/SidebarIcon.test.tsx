import React from "react"
import { mount, shallow } from "enzyme"
import SidebarIcon, { IconType } from "./SidebarIcon"
import { ResourceStatus } from "./types"
import { Color } from "./style-helpers"

const cases: Array<[string, ResourceStatus, Color]> = [
  ["isPending", ResourceStatus.Pending, Color.white],
  ["isHealthy", ResourceStatus.Healthy, Color.green],
  ["isUnhealthy", ResourceStatus.Unhealthy, Color.red],
  ["isBuilding", ResourceStatus.Building, Color.white],
  ["isWarning", ResourceStatus.Warning, Color.yellow],
  ["isNone", ResourceStatus.None, Color.white],
]

test.each(cases)("renders correctly - %s", (className, status, color) => {
  // TODO(han) - need to check that background-color is as expected
  // Since we're using styled-components, we should test individual style rules with
  // jest-styled-components, which requires an upgrade to styled-components v5
  const wrapper = shallow(<SidebarIcon status={status} alertCount={0} />)
  expect(wrapper).toMatchSnapshot()
})
