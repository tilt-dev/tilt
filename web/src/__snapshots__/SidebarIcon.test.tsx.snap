// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly - isBuilding 1`] = `
<styled.div
  className="isBuilding"
>
  <span>
     
  </span>
</styled.div>
`;

exports[`renders correctly - isHealthy 1`] = `
<styled.div
  className="isHealthy"
>
  <span>
     
  </span>
</styled.div>
`;

exports[`renders correctly - isNone 1`] = `
<styled.div
  className="isNone"
>
  <span>
     
  </span>
</styled.div>
`;

exports[`renders correctly - isPending 1`] = `
<styled.div
  className="isPending"
>
  <span>
     
  </span>
</styled.div>
`;

exports[`renders correctly - isUnhealthy 1`] = `
<styled.div
  className="isUnhealthy"
>
  <span>
     
  </span>
</styled.div>
`;

exports[`renders correctly - isWarning 1`] = `
<styled.div
  className="isWarning"
>
  <span>
     
  </span>
</styled.div>
`;
